File: app/local_plugins/sts_volcengine/android/src/main/kotlin/com/aiagent/sts_volcengine/StsVolcenginePlugin.kt
package com.aiagent.sts_volcengine

import com.aiagent.plugin_interface.NativeServiceRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

class StsVolcenginePlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        // 火山引擎 STS 服务，注册多个 vendor 名
        val vendors = listOf("doubao", "volcengine", "bytedance")
        for (vendor in vendors) {
            NativeServiceRegistry.registerSts(vendor) { StsVolcengineService(context) }
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: app/local_plugins/vendors/sts_volcengine/ios/Classes/StsVolcenginePlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// Volcengine realtime speech-to-speech Flutter plugin entry.
///
/// Mirrors the Android `StsVolcenginePlugin`: registers vendors "doubao",
/// "volcengine" and "bytedance" against `StsVolcengineService`. The service
/// drives the binary dialogue protocol on
/// `wss://openspeech.bytedance.com/api/v3/realtime/dialogue`.
public class StsVolcenginePlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.sts_volcengine", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        let vendors = ["doubao", "volcengine", "bytedance"]
        for vendor in vendors {
            NativeServiceRegistry.shared.registerSts(vendor) { StsVolcengineService() }
        }
        os_log("Registered NativeStsService for vendors: %{public}@",
               log: log, type: .debug, vendors.description)
    }
}
